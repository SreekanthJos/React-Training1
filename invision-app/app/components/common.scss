
.logo{
    color:#f65261;
    font-size: 18px;
}

  .logo__title {
    font-family: 'Aller-light';

  }

  .logo__titlebold {
    font-family: 'Aller-bold';
  }

.header {
    position: absolute;
    right: 0;
    top: 0;
    left: 0;
    z-index: 1;
    display: flex;
    align-items: center;
    height: 48px;
    padding-left: 24px;
    background: transparent;
    text-align: left;
  }

  .footer {
    position: fixed;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 1;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 48px;
    background: transparent;
    color: #f65261;
    text-align: center;
  }
  
.button {
  padding: 10px 20px;
  border: none;
  border-radius: 4px;
  outline: none;
  font-size: 16px; 
  cursor: pointer;
}
 
  .button__control {
    background-color: #424242;
    color: #f65261;
  }

  .button__primary {
    padding: 16px 40px;
        background-color: #f65261;
        color: #fff;
  }