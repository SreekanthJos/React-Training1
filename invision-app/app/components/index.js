export * from './Header'
export * from './Footer'
export * from './Home'
export * from './addMovie'
export * from './search'
export * from './movieFilter'
export * from './movieFiterAndSort'
export * from './sortBy'
export * from './moviesFound'
export * from './movieCard'
export * from './movieList'