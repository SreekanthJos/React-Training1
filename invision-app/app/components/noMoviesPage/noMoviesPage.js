import React from 'react';
import { ErrorBoundary } from 'react-error-boundary';
import { AddMovieButton } from '../addMovieButton'
import { MovieFiterAndSort } from '../movieFiterAndSort';

import { Header } from '../Header';
import { Footer } from '../Footer';
import './noMoviesPage.scss';

export function NoMoviesPage(
    {
        searchState,
        setSearchState,
        fetchMovies,
        setOrderState,
        setGenreState
    }
) 
{
       return (
        <>
            <Header></Header>
            <main className='no-movies-container'>
                <AddMovieButton searchState={searchState} setSearchState={setSearchState} fetchMovies={fetchMovies} history={history} />

                <MovieFiterAndSort setOrderState={setOrderState} setGenreState={setGenreState} />

                <div className='no-movies-content'>
                    <p className='no-movies-title'>No Movie Found</p>
                </div>
            </main>
            <Footer></Footer>
        </>
    );
}
