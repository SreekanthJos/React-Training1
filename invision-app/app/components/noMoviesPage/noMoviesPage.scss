.no-movies-container {
    background-color: #232323;
    width: 1100px;
    padding: 0 50px 80px 50px;
    font-family: "HelveticaNeue-Light", "Helvetica Neue Light", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
}

.top-part {
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-bottom: 5px ridge #555555;
}

.no-movies-content {
    height: 400px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.no-movies-title {
    font-size: 44px;
    color: #fff;
}
