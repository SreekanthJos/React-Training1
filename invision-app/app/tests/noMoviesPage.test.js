
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { NoMoviesPage } from '../components/noMoviesPage';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
const mockStore = configureStore();
const store = mockStore({});

let props = {
  searchState: '',
  setSearchState: jest.fn(),
  fetchMovies: jest.fn(),
  setOrderState: jest.fn(),
  setGenreState: jest.fn()
};

describe('no movies page:', () => {
  test('should render No movies page', () => {
    const searchBar = TestRenderer.create(<Provider store={store}><NoMoviesPage {...props} /></Provider>);
    const json = searchBar.toJSON();
    expect(json).toMatchSnapshot();
  })
});