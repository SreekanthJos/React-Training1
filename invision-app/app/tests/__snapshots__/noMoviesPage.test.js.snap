// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no movies page: should render No movies page 1`] = `
Array [
  <header
    className="header"
  >
    <span
      className="logo"
    >
      <span
        className="logo__titlebold"
      >
        netflix
      </span>
      <span
        className="logo__title"
      >
        roulette
      </span>
    </span>
  </header>,
  <main
    className="no-movies-container"
  >
    <section
      className="addContianer"
    >
      <div
        className="add__content"
      >
        <div
          className="add-movie"
        >
          <input
            className="button button__control"
            onClick={[Function]}
            type="button"
            value="+ADD MOVIE"
          />
        </div>
        <div
          className="search"
        >
          <h2
            className="search__header"
          >
            FIND YOUR MOVIE
          </h2>
          <form
            onSubmit={[Function]}
          >
            <div
              className="search__controls"
            >
              <input
                className="search__input"
                id="txtSearch"
                onChange={[Function]}
                type="text"
                value=""
              />
              <input
                className="button button__primary"
                type="submit"
                value="SEARCH"
              />
            </div>
             
          </form>
        </div>
      </div>
    </section>
    <div
      className="list-controls"
    >
      <div
        className="filter-container"
      >
        <span
          className="filter-item"
          name="All"
          onClick={[Function]}
        >
          All
        </span>
        <span
          className="filter-item"
          name="Documentary"
          onClick={[Function]}
        >
          Documentary
        </span>
        <span
          className="filter-item"
          name="Comedy"
          onClick={[Function]}
        >
          Comedy
        </span>
        <span
          className="filter-item"
          name="Hrror"
          onClick={[Function]}
        >
          Hrror
        </span>
        <span
          className="filter-item"
          name="Crime"
          onClick={[Function]}
        >
          Crime
        </span>
      </div>
      <div
        className="sort-container"
      >
        <span
          className="sort-header"
        >
          SORT BY
        </span>
        <select
          className="sort-select"
          onChange={[Function]}
        >
          <option
            value="release_date"
          >
            release date
          </option>
          <option
            value="title"
          >
            title
          </option>
        </select>
      </div>
    </div>
    <div
      className="no-movies-content"
    >
      <p
        className="no-movies-title"
      >
        No Movie Found
      </p>
    </div>
  </main>,
  <footer
    className="footer"
  >
    <span
      className="logo"
    >
      <span
        className="logo__titlebold"
      >
        netflix
      </span>
      <span
        className="logo__title"
      >
        roulette
      </span>
    </span>
  </footer>,
]
`;
